README.md

```diff
@@ -1,49 +1,79 @@
 # MixGleam
 
-A mix plugin for compiling Gleam code!
+An Elixir archive that teaches `Mix` how to work with Gleam code and
+dependencies!
 
 ## Installation
 
 Install the Gleam compiler onto your machine. [Installation instructions can
 be found here here](https://gleam.run/getting-started/installing-gleam.html).
 
-Add `mix_gleam` to your mix project deps:
+Install or update the `MixGleam` archive from Hex:
```

Comment:
Why is it that this is an archive rather than a dev dep?

Reply:
> ### How might we test this? Could we update the test project?

We could probably do some testing similar to how Elixir tests Mix:
https://github.com/elixir-lang/elixir/tree/main/lib/mix/test

What should we test? Here are some ideas:
- A `gleam`-tooled Gleam dependency
- A `rebar3`-tooled Gleam dependency
- A `mix`-tooled Gleam dependency

> ### Why is it that this is an archive rather than a dev dep?

The `gleam.deps.get` task needs to be compiled and available before Mix's  `deps.get` task runs. What `gleam.deps.get` does at the moment, however, is a bit of a hack—creating `mix.exs` files for `gleam`-tooled projects and ensuring that all their subdeps are fetched. There is probably a cleaner way to achieve the goals of `gleam.deps.get` though. Once `:bare` deps' subdeps are fetched by Mix, I suspect the only thing the `gleam.deps.get` task handles that would still need to happen is to ensure that Mix knows where to find Gleam deps' and subdeps' compiled Erlang artifacts—whether they're present to begin with or need to be compiled first. Adding the `aliases` to the main project's `mix.exs` might handle the rest for deps and subdeps too. https://github.com/gleam-lang/mix_gleam/pull/7/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R39

Additionally, the `gleam.compile` task needs to be compiled and available before Mix does any other compilation, which is not guaranteed, since other deps with Gleam code won't necessarily depend on `MixGleam`—particularly if they aren't `mix`-tooled projects.

https://github.com/gleam-lang/mix_gleam/pull/7/files#diff-54d6d3dbc67a8ec530426c6c47a22fe272d3edaa2911a0281ce515d133aefcb9R22-R26

Furthermore, once the `gleam.new` task is fully developed, there's no other option but for `MixGleam` to be an Elixir archive.

Reply:
RE testing: I would be happy with the test project being updated and running on CI, all else would be a bonus. I think we'll change this a bit with the next release so being things fairly fluid could be good use of time now.

>  once the gleam.new task is fully developed, there's no other option but for MixGleam to be an Elixir archive.

Assuming Mix is updated to download all the deps could this be used as a dep? Assuming they've already got a project and so they don't want to use the new command.

Archives add an extra step to run, and they slow down mix, so I think I'd rather have the new command being an archive but the rest being a dep. It would also make versioning easier.

Reply:
> Assuming Mix is updated to download all the deps could this be used as a dep?

As a manager, Gleam could probably call out to the Erlang manager at present, and even that shouldn't be needed with the next `compile-package` version.

With `:gleam` in the Project's list of `:compilers`, and Mix properly fetching subdeps, then probably `MixGleam` doesn't need to be an archive.

It's something we'd have to test. In my testing, I was unable to get it working except as an archive, but I had to start from the `deps.get` problem.

The current approach is one of several that could possibly fulfill our needs, and it's built for the current state of things, with some thought given to the future. I'm not sure the use of `aliases` in particular would work without `MixGleam` as an archive, so that would need adjusting and different approaches would have to be explored.

At present, `MixGleam` can use `gleam compile-package` to build any type of dep with `*.gleam` files in it; at least that's the goal.